File: payment/yoco.py
import requests
import os
from dotenv import load_dotenv

load_dotenv()

Secret_Key = os.environ['yoco_SECRET_KEY']

def yocoPayment(token,amount):

    response = requests.post(
    'https://online.yoco.com/v1/charges/',
    headers={
        'X-Auth-Secret-Key': Secret_Key,
        },
    json={
        'token': token,
        'amountInCents': amount,
        'currency': 'ZAR',
        },
    )

    if response.status_code == 201:
        return response.content
    else:
        return {"msg":"error"}

File: static/yoco.js

var yoco = new window.YocoSDK({
    publicKey: 'pk_test_ed3c54a6gOol69qa7f45',
  });
  

  window.onload = function myYoco(money) {

    yoco.showPopup({
      amountInCents: money,
      currency: 'ZAR',
      name: 'MY STORE',
      description: 'Awesome description',
      callback: function (result) {
        // This function returns a token that your server can use to capture a payment
        if (result.error) {
          const errorMessage = result.error.message;
          alert("error occured: " + errorMessage);
        } else {
            alert("card successfully tokenised: " + result.id);
            fetch('/pay', {
                    method: 'post',
                    headers: {
                        'Content-Type': 'application/json'
                    },
                    body: JSON.stringify({ token: result.id })
                }).then(res => res.json())
                   .then(data => {
                        if(data.errorCode){
                            alert(data.displayMessage);
                        }
                        else{
                            alert(data.status);
                        }
                    }).catch(error => {
                        alert(error.message);
                    })
        }

      }
    })
  };